di/container.go

```diff
@@ -0,0 +1,104 @@
+package di
+
+import (
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/go-playground/validator/v10/non-standard/validators"
+	"github.com/labstack/echo/v4"
+	"github.com/pkg/errors"
+	"github.com/proemergotech/log/v3"
+	"github.com/proemergotech/log/v3/echolog"
+
+	"github.com/artofimagination/mysql-resources-db-go-service/config"
+	"github.com/artofimagination/mysql-resources-db-go-service/dbcontrollers"
+	"github.com/artofimagination/mysql-resources-db-go-service/rest"
+	"github.com/artofimagination/mysql-resources-db-go-service/service"
+	"github.com/artofimagination/mysql-resources-db-go-service/validation"
+)
+
+type Container struct {
```

Comment:
If you want to use differenc comm interfaces in the same time, then yes, it supports this too, but I'm not sure in the way you think :D
I you would like to "choose/switch" between implementations you still need to add the other options into this container and you could start the proper interface within the `initialization` package. Where the container is initialized and the proper comm interface is started.

Reply:
If I get to change the protocol, I still want to keep REST as an alternative, so if a developer prefers gRPC or REST, they can freely choose, since both implementations are accessible.  but yes, You pretty much answered. I would need to add only a new layer that calls the "dbcontroller" functions